polaris-react/src/components/ChoiceList/ChoiceList.tsx

```diff
@@ -65,7 +64,8 @@ export function ChoiceList({
   // see https://github.com/Microsoft/TypeScript/issues/28768
   const ControlComponent: any = allowMultiple ? Checkbox : RadioButton;
 
-  const name = useUniqueId('ChoiceList', nameProp);
+  const uniqName = useId();
+  const name = nameProp ?? uniqName;
```

Comment:
This pattern is repeated across a couple of components. I think we could remove the `id` prop all together from some. I think we need to keep this property when there is a `<Label>` with a form element and we need to link the `htmlFor` to the `id`.